### ISSUE TITLE
Add package adds uninstalled package as being installed

### ISSUE THREAD

**Issue by timstoop** [aelsabbahy/goss#10]
A copy/paste says more than a thousand words:

```
vagrant@ceph1:/vagrant$ goss a package vim-tiny
Adding Package to './goss.json':

{
    "vim-tiny": {
        "installed": true,
        "versions": [
            ""
        ]
    }
}

vagrant@ceph1:/vagrant$ dpkg -l vim-tiny 
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                   Version          Architecture     Description
+++-======================-================-================-==================================================
un  vim-tiny               <none>           <none>           (no description available)
vagrant@ceph1:/vagrant$
```

I'd expect `goss a package` to recognise when a package is not installed. Or am I missing something?

**Comment by timstoop** [comment]
It actually works for packages that have never been installed:

```
vagrant@ceph1:/vagrant$ goss a package ibritish
Adding Package to './goss.json':

{
    "ibritish": {
        "installed": false
    }
}

vagrant@ceph1:/vagrant$ dpkg -l ibritish
dpkg-query: no packages found matching ibritish
vagrant@ceph1:/vagrant$
```

**Comment by timstoop** [comment]
The validation suffers from the same problem, an uninstalled package is seen as being installed even when it's not.

**Comment by aelsabbahy** [comment]
Seems like a misunderstanding on my part on how `dpkg-query` worked.

I was marking a package as installed/not based on exit status.

I'll change the logic to also check "${state}." I'll also add Debian integration-tests (I've been pushing this off).

Thanks for the copy/paste, very clear.

**Comment by timstoop** [comment]
Thanks. Just forked and looking at learning Go to see if I can help with these kind bugs.

**Comment by aelsabbahy** [comment]
Fixed in `v0.0.10`

**Comment by aelsabbahy** [comment]
Thanks for the feedback, let me know if you have any other issues.
